src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -196,7 +201,12 @@ export class FacetValueSuggestions extends Component {
   private async getQuerySuggestionsKeywords(omnibox: Omnibox): Promise<string[]> {
     if (this.options.useQuerySuggestions && omnibox.suggestionAddon) {
       const suggestions = (await omnibox.suggestionAddon.getSuggestion()) || [];
-      return suggestions.map(s => s.text);
+      return suggestions.map(({ html, text }) => {
+        if (text && text.length > 0) {
```

Comment:
While grouping two actions in a single map is more performant, we are usually only dealing with a very small number of suggestions (~10?) so I prefer to do two loops to have a clear separation of functions, and to remove the need for the reset statement above. e.g.

```
this.suggestionHTMLRepresentation = suggestions.reduce((obj, ({ html, text })) => {
    obj[suggestion.text] = html ? html : DomUtils.highlightElement(text, omnibox.getText());
}, {})

return suggestions.map(s => s.text);
```

I am curious why we would not get the `html` for a suggestion? Is the fallback of `DomUtils.highlightElement(text, omnibox.getText());` needed?

Reply:
Good point, I'll change the map for your reduce.

As for not getting the `html`, it's because the method passes a `string` to `FacetValueSuggestions.getFacetValueSuggestions`, which then passes it to a stack of sub-methods that all require a simple string. I was hesitant to change it, but both you and @francoislg seem to advise it, so I'll change it, unless told otherwise.

I added this fallback because I saw that `html` was optional, however, I haven't actually found any case where `html` is null. I could remove this fallback and run unit tests.

Reply:
Nice, even if it takes some more time, it's worth investing to see if `html` is always defined. There is a lot of value in eliminating if/else code-paths where one condition is a dead path the is never triggered. It makes the code easier to read/understand, test and maintain.

Reply:
Definitely! I fixed it now.